src/diffusers/pipelines/stable_diffusion/pipeline_flax_stable_diffusion_img2img.py

```diff
@@ -0,0 +1,414 @@
+
+        max_length = prompt_ids.shape[-1]
+        uncond_input = self.tokenizer(
+            [""] * batch_size, padding="max_length", max_length=max_length, return_tensors="np"
+        )
+        uncond_embeddings = self.text_encoder(uncond_input.input_ids, params=params["text_encoder"])[0]
+        context = jnp.concatenate([uncond_embeddings, text_embeddings])
+
+        # Create init_latents
+        init_latent_dist = self.vae.apply({"params": params["vae"]}, init_image, method=self.vae.encode).latent_dist
+        init_latents = init_latent_dist.sample(key=prng_seed).transpose((0, 3, 1, 2))
+        init_latents = 0.18215 * init_latents
+        latents_shape = (batch_size, self.unet.in_channels, height // 8, width // 8)
+        noise = jax.random.normal(prng_seed, shape=latents_shape, dtype=self.dtype)
+
+        def loop_body(step, args):
+            latents, timesteps, scheduler_state = args
+            # For classifier free guidance, we need to do two forward passes.
+            # Here we concatenate the unconditional and text embeddings into a single batch
+            # to avoid doing two forward passes
+            latents_input = jnp.concatenate([latents] * 2)
+
+            t = jnp.array(timesteps, dtype=jnp.int32)[step]
+            timestep = jnp.broadcast_to(t, latents_input.shape[0])
+
+            latents_input = self.scheduler.scale_model_input(scheduler_state, latents_input, t)
+
+            # predict the noise residual
+            noise_pred = self.unet.apply(
+                {"params": params["unet"]},
+                jnp.array(latents_input),
+                jnp.array(timestep, dtype=jnp.int32),
+                encoder_hidden_states=context,
+            ).sample
+            # perform guidance
+            noise_pred_uncond, noise_prediction_text = jnp.split(noise_pred, 2, axis=0)
+            noise_pred = noise_pred_uncond + guidance_scale * (noise_prediction_text - noise_pred_uncond)
+
+            # compute the previous noisy sample x_t -> x_t-1
+            latents, scheduler_state = self.scheduler.step(scheduler_state, noise_pred, t, latents).to_tuple()
+            return latents, timestep, scheduler_state
+
+        scheduler_state = self.scheduler.set_timesteps(
+            params["scheduler"], num_inference_steps=num_inference_steps, shape=latents_shape
+        )
+
+        timesteps = self.get_timesteps(num_inference_steps, strength, scheduler_state)
+        latent_timestep = timesteps[:1].repeat(batch_size)
+        init_latents = self.scheduler.add_noise(init_latents, noise, latent_timestep)
+        latents = init_latents
+
+        if debug:
+            # run with python for loop
+            for i in range(len(timesteps)):
+                latents, timesteps, scheduler_state = loop_body(i, (latents, timesteps, scheduler_state))
+        else:
+            latents, _, _ = jax.lax.fori_loop(0, len(timesteps), loop_body, (latents, timesteps, scheduler_state))
```

Comment:
I'm not sure about this. Wouldn't it be possible to start the loop at the initial effective timestep (instead of 0), and avoid having to pass it around?

Reply:
This is also done. Please have a look when you can